packages/core-mobile/app/services/fcm/AppCheckService.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  firebase,
+  FirebaseAppCheckTypes
+} from '@react-native-firebase/app-check'
+import Logger from 'utils/Logger'
+import Config from 'react-native-config'
+
+class AppCheckService {
+  init = (): void => {
+    const rnfbProvider = firebase
+      .appCheck()
+      .newReactNativeFirebaseAppCheckProvider()
+    rnfbProvider.configure({
+      android: {
+        provider: __DEV__ ? 'debug' : 'playIntegrity',
+        debugToken: Config.APPCHECK_DEBUG_TOKEN
+      },
+      apple: {
+        provider: __DEV__ ? 'debug' : 'appAttestWithDeviceCheckFallback',
+        debugToken: Config.APPCHECK_DEBUG_TOKEN
+      }
+    })
+
+    firebase
+      .appCheck()
+      .initializeAppCheck({
+        provider: rnfbProvider,
+        isTokenAutoRefreshEnabled: true
+      })
+      .catch(reason => {
+        Logger.error(`initializeAppCheck failed: ${reason}`)
+      })
+  }
+
+  /**
+   * For debug purpose, to check if setup is correct
+   */
+  verifyAppCheck = async (): Promise<void> => {
+    try {
+      const { token } = await this.getToken()
+
+      if (token.length > 0) {
+        Logger.info('AppCheckService:verifyAppCheck passed')
+      }
+    } catch (error) {
+      Logger.error(`[AppCheckService.ts][verifyAppCheck]${error}`)
+    }
+  }
+
+  getToken = async (): Promise<FirebaseAppCheckTypes.AppCheckTokenResult> => {
+    return await firebase.appCheck().getToken(false)
+  }
+
+  fetch = async (url: string, bodyJson: string): Promise<Response> => {
+    const appCheckToken = await this.getToken()
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Firebase-AppCheck': appCheckToken.token
+      },
+      body: bodyJson
+    }
+    return fetch(url, options)
+  }
```

Comment:
can we move this fetch to a HTTP client service/util? app check is just the one that we use right now but it can be others as well.

Reply:
But this is fetch that puts appCheck token into header, it's not used by appcheck but anything can use it if they need to use appCheck header. I think it's gonna be easier to find in AppCheckService than when hidden somewhere in general place.

Reply:
we need a http client/util anyway and I believe we are going to utilize app check going forward for other backend apis

Reply:
[Extract fetchWithAppCheck](https://github.com/ava-labs/core-mobile/pull/1659/commits/b06cb4b179e00369baaa83b3334a244323aab560)